dagql/server.go

```diff
@@ -467,26 +467,8 @@ func (s *Server) Load(ctx context.Context, id *call.ID) (Object, error) {
 	} else {
 		base = s.root
 	}
-	astField := &ast.Field{
-		Name: id.Field(),
-	}
-	vars := map[string]any{}
-	for _, arg := range id.Args() {
-		vars[arg.Name()] = arg.Value().ToInput()
-		astField.Arguments = append(astField.Arguments, &ast.Argument{
-			Name: arg.Name(),
-			Value: &ast.Value{
-				Kind: ast.Variable,
-				Raw:  arg.Name(),
-			},
-		})
-	}
-	sel, _, err := base.ObjectType().ParseField(ctx, id.View(), astField, vars)
-	if err != nil {
-		return nil, fmt.Errorf("parse field %q: %w", astField.Name, err)
-	}
-	sel.Nth = int(id.Nth())
-	res, id, err := s.cachedSelect(ctx, base, sel)
```

Comment:
This chunk of code is why `cachedSelect` changed from accepting a Selector to accepting an ID.

This code in `Load` was taking an ID, converting it into a `Selector`, only for it to go back to an ID in `cachedSelect`. I think this was a bit wasteful from a performance perspective but more importantly it broke custom digests since the Selector didn't know about those and instead just always re-calculated the ID digest from the call chain.

I fixed it by just providing the ID directly since that seems simpler overall. The other callers of `cachedSelect` (now renamed to `cachedCall`) are barely impacted since they start with a Selector but can easily turn that into an ID by `Selector.AppendTo`